main/network.c

```diff
@@ -859,54 +859,48 @@ php_socket_t php_network_connect_socket_to_host(const char *host, unsigned short
 		if (sa) {
 			/* make a connection attempt */
 
-			if (bindto) {
-				struct sockaddr *local_address = NULL;
-				int local_address_len = 0;
-
+			while (bindto) {
+				union {
+					struct sockaddr common;
+					struct sockaddr_in in4;
+#if HAVE_IPV6 && HAVE_INET_PTON
+					struct sockaddr_in6 in6;
+#endif
+				} local_address = { 0 };
```

Comment:
Now that the bind call is in each branch, do we need this union at all? Can't you now directly declare `struct sockaddr_in` and `struct sockaddr_in6` respectively in the branches?

Reply:
Right. I am completely caught in the trap of the original code...